### PR TITLE
feat: Introduce `kraft pkg prune`

### PR DESCRIPTION
**Pull request by mdsahiloss**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Tested your changes against relevant architectures and platforms;
  - [ ] Updated relevant documentation.

### Description of changes
Added subcommand remove for the command kraft pkg that removes the pulled package locally on the system.

### CODE REVIEW

File: cmd/kraft/pkg/prune/prune.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		Short:   "Prune unikraft package available on the disc",
+		Use:     "prune [FLAGS] [PACKAGE|DIR]",
+		Aliases: []string{"pr"},
+		Long: heredoc.Doc(`
+		Prunes unikraft package available locally on the host in the directory $HOME/.local/share/kraftkit/sources
+		`),
+		Example: heredoc.Doc(`
+			# Prunes unikraft package nginx.
+			$ kraft pkg prune nginx
+			$ kraft pkg prune nginx:stable`),
```
Comment (craciunoiuc) [inline cmd/kraft/pkg/prune/prune.go]: I would add more examples here. For example also the case where OCI packages are used.

Comment (craciunoiuc) [reply cmd/kraft/pkg/prune/prune.go]: Every stubbed `Delete()` function below should have an equivalent example here

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey

Took a brief look and the command looks fine.
Try to also implement some of the `Delete` functions. `Manifest` ones should be good to start with. `Directory` afterwards.

File: manifest/pack.go [hunk 2]
```diff
@@ -98,6 +100,26 @@ func (mp mpack) Pull(ctx context.Context, opts ...pack.PullOption) error {
...
+	localAvailablePackages, err := os.ReadDir(cacheDir)
+	if err != nil {
+		return err
+	}
+	for _, localPackage := range localAvailablePackages {
+		if strings.Contains(localPackage.Name(), mp.Name()) && strings.Contains(localPackage.Name(), mp.Version()) {
+			if !strings.HasSuffix(cacheDir, "/") {
+				cacheDir += "/"
+			}
+			err = os.Remove(cacheDir + localPackage.Name())
```
Comment (craciunoiuc) [inline manifest/pack.go]: You can use functions from the `filepath` package to work with paths

File: packmanager/umbrella.go [hunk 3]
```diff
@@ -123,6 +126,43 @@ func (u umbrella) AddSource(ctx context.Context, source string) error {
 	return nil
 }
 
+func (u umbrella) Prune(ctx context.Context, qopts ...QueryOption) error {
+	var query Query
+	sourcesDir := config.G[config.KraftKit](ctx).Paths.Sources
+	if !strings.HasSuffix(sourcesDir, "/") {
+		sourcesDir += "/"
+	}
```
Comment (craciunoiuc) [inline packmanager/umbrella.go]: Same here

File: manifest/pack.go (hunk 2 above)
Comment (mdsahiloss) [reply manifest/pack.go]: Done 👍🏽

File: packmanager/umbrella.go (hunk 3 above)
Comment (mdsahiloss) [reply packmanager/umbrella.go]: Done 👍🏽

File: cmd/kraft/pkg/prune/prune.go (hunk 1 above)
Comment (mdsahiloss) [reply cmd/kraft/pkg/prune/prune.go]: @craciunoiuc actually `Delete()` is currently only implemented for manifest package not for OCIPackage and ELFLoader.

File: oci/pack.go [hunk 4]
```diff
@@ -520,6 +521,44 @@ unpack:
...
+
+	files, err := os.ReadDir(path.Join(ociDir, "manifests", ocipack.Name()))
+	if err != nil {
+		return err
+	}
+	if len(files) == 0 {
+		if err = os.RemoveAll(path.Join(ociDir, "manifests", ocipack.Name())); err != nil {
+			return err
+		}
+	}
```
Comment (craciunoiuc) [inline oci/pack.go]: I think that the use of `ocipack.Name()` is incorrect here. My files were not deleted completely:

```
├── configs
│   └── sha256
├── layers
│   └── sha256
└── manifests
    ├── index.unikraft.io
    │   └── unikraft.org
    │       └── nginx
    │           └── latest.json
    └── unikraft.org
```

Comment (mdsahiloss) [reply oci/pack.go]: @craciunoiuc Now give it a try, Hoping will work fine :)

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**